communication/src/build.mk

```diff
@@ -13,7 +13,7 @@ TARGET_SRC_PATH = src
 # C++ source files included in this build.
 CPPSRC += $(TARGET_SRC_PATH)/coap.cpp
 CPPSRC += $(TARGET_SRC_PATH)/handshake.cpp
-CPPSRC += $(TARGET_SRC_PATH)/spark_protocol.cpp
+CPPSRC += $(TARGET_SRC_PATH)/core_protocol.cpp
```

Comment:
That implementation only used in the Core. It can be confusing to see SparkProtocol and Protocol instances and not know which one to use. (Zak hit this issue.) . Naming it CoreProtocol seemed to be more descriptive, although a comment in the class header would also have been useful. :-)

Reply:
But the class is still called `SparkProtocol`, it's just the file that got renamed

Reply:
Yeah...we should probably rename the class too and add a comment about how it's used and that new functionality should not be added.

Reply:
I'd suggest to keep everything under original names, and once the Core is finally and officially EOL'ed remove the old implementation from the source tree altogether

Reply:
I can confirm this creates a lot of friction for a new person. It should be easier for those familiar with the code base to follow the change, than those who are unfamiliar with the code base to understand the ambiguous naming scheme.

Reply:
I will be extending the changes to make the `SparkProtocol` --> `CoreProtocol`.

Reply:
Let's keep only relevant changes in this PR. The class can be renamed in a separate PR if necessary